examples/counter/web-leptos/src/main.rs

```diff
@@ -0,0 +1,118 @@
+mod http;
+mod sse;
+
+use std::rc::Rc;
+
+use anyhow::Result;
+use futures_util::TryStreamExt;
+use leptos::{
+    component, create_effect, create_signal, spawn_local, view, IntoView, Scope, SignalGet,
+    SignalUpdate, WriteSignal,
+};
+use serde::{Deserialize, Serialize};
+use shared::{App, Capabilities, Core, Effect, Event, ViewModel};
+
+// we need a newtype for Event because signals require Clone, so use json instead
+#[derive(Debug, Serialize, Deserialize)]
+struct Task(Event);
+
+impl Task {
+    fn to_json(&self) -> String {
+        serde_json::to_string(self).unwrap()
+    }
+
+    fn from_json(json: &str) -> Result<Self> {
+        serde_json::from_str(json).map_err(Into::into)
+    }
+}
```

Comment:
This is a bit yuck.  Guessing you can't just use an `Rc` because you need to take it out of the `Rc` again at some point?

Also, don't we `serde(skip)` some variants - so those won't work with this example?

Reply:
Would love an alternative to this. So will have a think over the weekend. Unless you have immediate ideas. The `serde(skip)` is on events that stay inside the core, so shouldn't get involved. We just need a way to represent the `Event` that implements the traits leptos signals need (off the top of my head `Clone`, `PartialEq, `Eq`, maybe `Hash`) etc and we can't add those to `Event` right now because of the Capabilities' Events (which ironically aren't involved). Wonder if we should look at splitting the Event into internal and external, and adding all those trait impls to the external set?

Reply:
Right I see - I didn't really understand what was/wasn't going through this `Task` thing.

In that case you _could_ make this hack more elegant (but also more dangerous):

```
impl Clone for Task {
    fn clone(&self) -> Task {
        serde_json::from_str(serde_json::to_string(self).unwrap()).unwrap()
    }
}
```

Reply:
Although yeah it would be nice to find a more elegant solution.  

Splitting `Event` would work but would it be nice?  I'm not sure.

Maybe we could also look into forcing `Capabilities` events to impl the various traits?  I forget the reasons why it was difficult... some `http` type not being `Clone` was it?  can we just roll our own?  dunno

Reply:
Thanks @obmarg ! I implemented `Clone` on `Task` for now — you're right it's nicer. I'll have a play with implementing `Clone` on `Event` and report back.

Reply:
OK, my bad. It works fine implementing `Clone` on `Event`. I was also assuming it was the HTTP stopping us. But happily not.